packages/serverpod/lib/src/database/adapters/postgres/database_connection.dart

```diff
@@ -200,7 +210,13 @@ class DatabaseConnection {
         'UPDATE "${table.tableName}" AS t SET $setColumns FROM (VALUES $values) AS data($columnNames) WHERE data.id = t.id RETURNING *';
 
     var result =
-        await _mappedResultsQuery(session, query, transaction: transaction);
+        (await _mappedResultsQuery(session, query, transaction: transaction))
+            .toList();
+
+    result = _mergeResultsWithNonPersistedFields(
+      rows: rows,
+      dbResults: result,
+    );
```

Comment:
Suggestion: Maybe this code (and the same code earlier) can be made a bit more functional to make it more declarative and easier to follow. Something along these lines should be possible:
```dart
    return (await _mappedResultsQuery(session, query, transaction: transaction))
        .map(_mergeResultWithNonPersistedFields(rows))
        .map(_poolManager.serializationManager.deserialize<T>)
        .toList();
```

Then `_mergeResultWithNonPersistedFields` would be something like:
```dart
  Map<String, dynamic> Function(T)
      _mergeResultWithNonPersistedFields<T extends TableRow>(
    Map<String, dynamic> dbResult,
  ) {
    return (T row) => {
          // Add non-persisted fields from the original object
          ...row.toJson(),
          // Override with database fields (common fields)
          ...dbResult,
        };
  }
```

Reply:
Thanks for this, done!